OWML.ModHelper.Input/ModInputHandler.cs

```diff
@@ -0,0 +1,375 @@
+        {
+            return (!(IsPressed_Combo(combination) || IsPressed_Single(combination)))
+                && ((ModInputCombination)combination).IsRelevant(_tapKeep);
+        }
+
+        private KeyCode StringToKeyCode(string key)
+        {
+            key = key.Trim();
+            if (key.Contains(xboxPrefix))
+            {
+                var xboxKey = key.Substring(xboxPrefix.Length);
+                var xboxCode = (XboxButton)Enum.Parse(typeof(XboxButton), xboxKey, true);
+                if (Enum.IsDefined(typeof(XboxButton), xboxCode))
+                {
+                    return InputTranslator.GetKeyCode(xboxCode, false);
+                }
+                return KeyCode.None;
+            }
+            else
+            {
+                var changedKey = key;
+                if (key == "control" || key == "ctrl")
+                {
+                    return KeyCode.LeftControl;
+                }
+                if (key == "shift")
+                {
+                    return KeyCode.LeftShift;
+                }
+                if (key == "alt")
+                {
+                    return KeyCode.LeftAlt;
+                }
+                var code = (KeyCode)Enum.Parse(typeof(KeyCode), changedKey, true);
+                if (Enum.IsDefined(typeof(KeyCode), code))
+                {
+                    return code;
+                }
+                return KeyCode.None;
+            }
+        }
+
+        private long ParseCombination(string combo, bool forRemoval = false)
+        {
+            var thisCombination = new int[_maxComboLength];
+            var i = 0;
+            foreach (var key in combo.Trim().ToLower().Split('+'))
+            {
+                if (i >= _maxComboLength)
+                {
+                    return (int)RegistrationCode.CombinationTooLong;
+                }
+                KeyCode code = StringToKeyCode(key);
+                if (code == KeyCode.None)
+                {
+                    return (int)RegistrationCode.InvalidCombination;
+                }
+                thisCombination[i] = (int)code;
+                i++;
+            }
+            Array.Sort(thisCombination);
+            long hash = 0;
+            for (i = 0; i < _maxComboLength; i++)
+            {
+                hash = hash * _maxUsefulKey + thisCombination[i];
+            }
+            if (hash < _maxUsefulKey && _gameBindingCounter[hash] > 0)
+            {
+                return (int)RegistrationCode.CombinationTaken;
+            }
+            return (_comboRegistry.ContainsKey(hash) && !forRemoval ? (int)RegistrationCode.CombinationTaken : hash);
+        }
+
+        public RegistrationCode RegisterCombination(IModInputCombination combination)
+        {
+            var castCombo = (ModInputCombination)combination;
+            castCombo.ClearSingles();
+            if (castCombo == null || castCombo.Combo == null)
```

Comment:
castCombo  can't be null (here), it would've crashed on the line above

Reply:
fixed